docs/building.md

```diff
@@ -1,121 +1,120 @@
-git submodule init
-git submodule update --recursive
-```
-
-The following command will ensure your environment variables are set properly and build the dependencies needed by the project:
-
-```
-./libs/verify-desktop-environment.sh
-```
-
-You can then run the Rust tests using:
-
-```
-cargo test
-```
-
-## Android development
-
-### Java 8
-
-Using any other version than 8 __will__ mess up your build and is not supported.  
-Please ensure `JAVA_HOME` points to the JDK 8 installation directory.
-
-### Setting up the Android SDK
-
-#### Through Android Studio
-
-1. Download and install Android Studio and let it install the SDK for you.
-1. In the "SDK Manager" window, find out the location of the SDK.
-1. Set `ANDROID_SDK_ROOT` and `ANDROID_HOME` (technically deprecated 🤷‍♂️) to this location and add it to your rc file.
-
-#### Manually
-
-Visit https://developer.android.com/studio/ and download the "Command line tools".  
-Install the SDK by executing:
-
-
-```
-> cd ~  
-> mkdir -p android-sdk/cmdline-tools  
-> cd android-sdk/cmdline-tools  
-> unzip {path-to.zip}  
-> # Don't forget to write these to your rc file!
-> export ANDROID_SDK_ROOT=$HOME/android-sdk
-> export ANDROID_HOME=$ANDROID_SDK_ROOT
-> $ANDROID_SDK_ROOT/cmdline-tools/tools/bin/sdkmanager --licenses
-```
-
-### Rust targets, NDK
-
-After the Android SDK is installed, run the following script that will ll install the necessary Rust targets, the Android NDK and has built-in and helpful messages to ensure you will be able to compile for Android properly:
-
-```
-./libs/verify-android-environment.sh
-````
-
-You may have to run the above script multiple times till it succeeds, once it does you can try building using:
-
+When working on application-services Rust Components it's important to set up your environment for both building the application-services library, but also for Android or iOS as appropriate to test your changes in our clients.
+
+## First time builds
+
+Building for the first time is more complicated than a typical Rust project.
+To build for an end-to-end experience that enables you to test changes in our
+client applications like Fenix and Firefox iOS, there are a number of build
+systems required for all the dependencies. The initial setup is likely to take
+a number of hours to complete.
+
+
+## Building the Rust Components
+
+*Complete this section before moving to the android/iOS build instructions.*
+
+1. Install Rust
+   1. Install Rust: [via rustup](https://www.rust-lang.org/tools/install)
+   1. [set your version](https://doc.rust-lang.org/edition-guide/rust-2018/rustup-for-managing-rust-versions.html) to match the version specified in the [rust-toolchain](../rust-toolchain) file.
+   1. Run: `rustc --version` to confirm.
+1. Install your system dependencies - follow the instructions below for Linux, MacOS or Windows
+1. Install the application-services required submodules
+   1. Run: `git submodule init`
+   1. Run: `git submodule update --recursive`
```

Comment:
I think we should move the submoules into a "getting the code" section, wherever that is; they're more "git things" than "dependency things".